pkg/collector/runner/check_stats.go

```diff
@@ -0,0 +1,118 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package runner
+
+import (
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	// How long is the first series of check runs we want to log
+	firstRunSeries uint64 = 5
+)
+
+var checkStats *runnerCheckStats
+
+func init() {
+	checkStats = &runnerCheckStats{
+		Stats: make(map[string]map[check.ID]*check.Stats),
+	}
+}
+
+// runnerCheckStats holds the stats from the running checks
+type runnerCheckStats struct {
+	Stats map[string]map[check.ID]*check.Stats
+	M     sync.RWMutex
```

Comment:
I think they can be private. Also the mutex can be a composition

```suggestion
	sync.RWMutex
	stats map[string]map[check.ID]*check.Stats   
```

Reply:
agreed, same question for agent core, happy to make the changes if they approve